packages/atomic/cypress/integration/search-box.cypress.ts

```diff
@@ -72,28 +90,40 @@ describe('Search Box Test Suites', () => {
 
     describe('with input', () => {
       const expectedSum = numOfSuggestions + numOfRecentQueries;
-      beforeEach(() => {
+
+      function setInputText() {
         SearchBoxSelectors.inputBox().type('Rec');
-      });
+      }
 
-      SearchBoxAssertions.assertHasSuggestionsCount(expectedSum);
-      CommonAssertions.assertAriaLiveMessage(
-        SearchBoxSelectors.liveRegion,
-        expectedSum.toString()
-      );
+      describe('verify rendering', () => {
+        before(() => {
+          setupWithSuggestionsAndRecentQueries();
+          setInputText();
+        });
+
+        SearchBoxAssertions.assertHasSuggestionsCount(expectedSum);
+        CommonAssertions.assertAriaLiveMessage(
+          SearchBoxSelectors.liveRegion,
+          expectedSum.toString()
+        );
+      });
 
       describe('after selecting a suggestion with the mouse', () => {
-        beforeEach(() => {
+        before(() => {
+          setupWithSuggestionsAndRecentQueries();
+          setInputText();
           SearchBoxSelectors.querySuggestions().eq(1).click();
-          cy.wait(TestFixture.interceptAliases.Search);
         });
 
         SearchBoxAssertions.assertFocusSearchBox();
         SearchBoxAssertions.assertHasText('Recent query 1');
       });
 
       describe('after focusing a suggestion with the keyboard', () => {
-        beforeEach(() => {
+        before(() => {
+          setupWithRecentQueries();
+          setInputText();
+
           SearchBoxSelectors.inputBox()
             .focus()
             .type('{downarrow}{downarrow}{downarrow}');
```

Comment:
On my end, recent queries appear first. This is strange.
![image](https://user-images.githubusercontent.com/54454747/149517404-2863bc14-f008-4b54-9459-3416df39f7a0.png)

Regardless, it takes two down arrows for me to reach the first recent query, not three. Maybe the down arrows are being pressed too fast or too early?

Reply:
The test setup has the reverse order for query suggestions and recent queries (see [here](https://github.com/coveo/ui-kit/blob/master/packages/atomic/cypress/integration/search-box-actions.ts#L21)).

The recent queries in the test use a 0-based index. So two `keydown` events arrive at `Recent Query 0`, and three reach `Recent Query 1`.

> Maybe after each down arrow, we should wait for the input text to be changed.

This would be ideal, but it means at least two `type` commands separated by either a wait or assertion? I [did not find a way](https://github.com/coveo/ui-kit/pull/1668/files#r785211857) to write such a chain, while also avoiding blurs. After trying different things, it felt like there was a constraint of needing to do everything in a single `type` command. Maybe I missed something?